### PR TITLE
Assignments

### PR DESCRIPTION
**Pull request by sureshHARDIYA**

### CODE REVIEW

File: backend/src/api/assignments/queries/assignmentsList.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const graphqlSelectRequestedAttributes = require('../../shared/utils/graphqlSelectRequestedAttributes');
+
+const schema = `
+  assignmentsList(filter: AssignmentsFilterInput, limit: Int, offset: Int, orderBy: AssignmentsOrderByEnum): AssignmentsPage!
+`;
+
+const resolver = {
+  assignmentsList: async (root, args, context, info) => {
+    new PermissionChecker(context).validateHas(
+      permissions.casedRead,
```
Comment (hunghkit) [inline backend/src/api/assignments/queries/assignmentsList.js:14]: are you sure `casedRead` permission?

File: backend/src/api/assignments/types/assignmentEnum.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const schema = `
+  enum questionEnum {
```
Comment (hunghkit) [inline backend/src/api/assignments/types/assignmentEnum.js:2]: Graphql enum, type is capitalize 
```
questionEnum -> QuestionEnum
```

File: frontend/src/modules/assignments/assignmentsSelectors.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { createSelector } from 'reselect';
+import authSelectors from 'modules/auth/authSelectors';
+import PermissionChecker from 'modules/auth/permissionChecker';
+import Permissions from 'security/permissions';
+
+const selectPermissionToRead = createSelector(
+  [authSelectors.selectCurrentUser],
+  (currentUser) =>
+    new PermissionChecker(currentUser).match(
+      Permissions.values.audioRead,
```
Comment (hunghkit) [inline frontend/src/modules/assignments/assignmentsSelectors.js:10]: audio permission?

File: frontend/src/modules/assignments/destroy/assignmentsDestroyReducers.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import actions from 'modules/assignments/destroy/assignmentsDestroyActions';
+
+const initialData = {
+  loading: false,
+};
+
+export default (state = initialData, { type, payload }) => {
+  if (type === actions.DESTROY_ALL_STARTED) {
```
Comment (hunghkit) [inline frontend/src/modules/assignments/destroy/assignmentsDestroyReducers.js:8]: you can use switch/case in here

File: frontend/src/modules/assignments/form/assignmentsFormReducers.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+};
+
+const initialData = {
+  findLoading: false,
+  saveLoading: false,
+  record: null,
+};
+
+export default (state = initialData, { type, payload }) => {
+  if (type === actions.RESET) {
```
Comment (hunghkit) [inline frontend/src/modules/assignments/form/assignmentsFormReducers.js:14]: pls move to switch/case

File: frontend/src/view/assignments/form/builder/FormBuilder/builder.js [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+  <SortableCard
+    onDelete={onDelete}
+    onChange={onChange}
+    index={index}
+    value={value}
+  />
+);
+
+const SortableSchema = SortableContainer(({ items, header, onChange }) => {
+  return (
```
Comment (hunghkit) [inline frontend/src/view/assignments/form/builder/FormBuilder/builder.js:23]: can reduce "return"

File: frontend/src/view/assignments/form/builder/FormBuilder/builder.js [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+        if(dispatch) {
+          dispatch(actions.doCreate(formData));
+        }
+      } else if (onError) {
+        setErrors(err.schema.errors);
+        onError(err);
+      }
+    });
+  };
+
```
Comment (hunghkit) [inline frontend/src/view/assignments/form/builder/FormBuilder/builder.js:184]: `initialValue` should be `record` not `formStructure `

File: frontend/src/view/assignments/form/builder/FormBuilder/builder.js [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+  const handleSubmit = (e) => {
+    setErrors([]);
+    e.preventDefault();
+    validateFields((err, formData) => {
+      if (!err) {
+        if (onSave) onSave(formData);
+
+        if(dispatch) {
+          dispatch(actions.doCreate(formData));
```
Comment (hunghkit) [inline frontend/src/view/assignments/form/builder/FormBuilder/builder.js:176]: Alway call doCreate, so you can't edit

**Review by hunghkit — CHANGES_REQUESTED** [review]

File: frontend/src/view/assignments/form/builder/FormBuilder/builder.js (hunk 6 above)
Comment (sureshHARDIYA) [reply frontend/src/view/assignments/form/builder/FormBuilder/builder.js]: What do you mean by this?

**Review by hunghkit — APPROVED** [review]

**Pull request merged**